### PR TITLE
Adding foundry compiler

### PR DESCRIPTION
**Pull request by BlueAlder**
Creating this base utility image for blockchain compilations.

### CODE REVIEW

File: vendor/ethcompiler/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+```dockerfile
+FROM ghcr.io/downunderctf/docker-vendor/ethcompiler:foundry as build
+
+COPY contracts/ src/
+COPY foundry.toml foundry.toml
+
+RUN /root/.foundry/bin/forge build
+
+FROM ghcr.io/downunderctf/eth-challenge-base-priv/eth-base:latest as dist
```
Comment (lecafard) [inline vendor/ethcompiler/README.md:16]: non public image

Comment (lecafard) [inline vendor/ethcompiler/README.md]: ig it's an example, not sure if you'd want to keep it here

File: vendor/ethcompiler/dockerfiles/Dockerfile.foundry [hunk 2]
```diff
@@ -0,0 +1,14 @@
+FROM ubuntu:jammy
```
Comment (lecafard) [inline vendor/ethcompiler/dockerfiles/Dockerfile.foundry]: nit: can you use version number instead of codename?

**Review by lecafard — APPROVED** [review]
one nit

File: vendor/ethcompiler/README.md (hunk 1 above)
Comment (BlueAlder) [reply vendor/ethcompiler/README.md]: Done

**Pull request merged**